src/main/scala/io/circe/yaml/Printer.scala

```diff
@@ -27,17 +28,49 @@ final case class Printer(
   version: YamlVersion = YamlVersion.Auto
 ) {
 
+  def containsBuggyChars(json: Json): Boolean = {
+    def isBad(s: String): Boolean =
+      s.indexOf('\u0085') >= 0 || s.indexOf('\ufeff') >= 0
+
+    @tailrec def loop(js: List[Json]): Boolean =
+      js match {
+        case h :: t =>
+          h match {
+            case Json.JArray(js) => loop(js.toList ::: t)
+            case Json.JObject(o) => loop(o.values.toList ::: t)
+            case Json.JString(s) if isBad(s) => true
+            case _ => loop(t)
+          }
+        case Nil =>
+          false
+      }
+    loop(json :: Nil)
+  }
+
   def pretty(json: Json): String = {
     val rootTag = yamlTag(json)
     val writer = new StringWriter()
-    val serializer = new Serializer(new Emitter(writer, options), new Resolver, options, rootTag)
+    val opts = if (containsBuggyChars(json)) safeOptions else options
```

Comment:
It'd be nice to skip this potentially expensive check if the options are already safe.